From b8e23426ec5ac97efdceaf2a319b9d5921e7b412 Mon Sep 17 00:00:00 2001
From: Patrick Nicolas <patricknicolas@laposte.net>
Date: Tue, 5 Aug 2025 16:01:59 +0200
Subject: [PATCH 3/3] c/util: Request video_maintenance_1 feature

This will only be enabled if the VK_KHR_maintenance1 extension was
requested.
---
 src/xrt/compositor/util/comp_vulkan.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/xrt/compositor/util/comp_vulkan.c b/src/xrt/compositor/util/comp_vulkan.c
index 5618f4f78..9c719275c 100644
--- a/src/xrt/compositor/util/comp_vulkan.c
+++ b/src/xrt/compositor/util/comp_vulkan.c
@@ -282,6 +282,7 @@ create_device(struct vk_bundle *vk, const struct comp_vulkan_arguments *vk_args)
 	    .timeline_semaphore = vk_args->timeline_semaphore,
 	    .synchronization_2 = true,
 	    .present_wait = true,
+	    .video_maintenance_1 = true,
 	};
 
 	ret = vk_init_mutex(vk);
-- 
2.50.1

